packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3415,11 +3415,15 @@ export class ContainerRuntime
 	 * @param messageTimestampMs - The timestamp of the message that added the reference.
 	 */
 	public addedGCOutboundRoute(fromPath: string, toPath: string, messageTimestampMs?: number) {
-		this.garbageCollector.addedOutboundReference(
-			fromPath,
-			toPath,
-			messageTimestampMs ?? this.getCurrentReferenceTimestampMs(),
+		// This is always called when processing an op so messageTimestampMs should exist. Due to back-compat
+		// across the data store runtime / container runtime boundary, this may be undefined and if so, get
+		// the timestamp from the last processed message which must exist.
```

Comment:
The only exception I can think of is if this is called when detached, but I couldn't find anywhere besides the alias one where that would be the case.  Once we get past n/n-1 for dataStoreRuntime we'll be all good for sure and can update the types.  Can you open a follow-up item for that?

Reply:
Maybe it will be safer to add an error log and return here. Then with the follow-up item, this will always be defined. No point in taking the risk.

Reply:
I opened a follow-up item already - AB#8329